### PR TITLE
Implement SPACE() and QUOTE() functions.

### PR DESCRIPTION
**Pull request by tntnatbry**
Adds new functionality and fixes test working_tpch1/postprocess/func_2.sql.

### CODE REVIEW

**Review by LinuxJedi — COMMENTED** [review]
I'm not sure this quote implementation will work with UTF-8 data. Can you please confirm?

**Comment by tntnatbry** [comment]
It should handle utf-8 data fine since we operate on one-byte chars of C++ string type.

Below is an example. As can be seen from the output of hex(), the 3rd row is a 2-byte character, and the quote() function works as expected.

```sql
MariaDB [(none)]> create database test2 character set = 'utf8';
Query OK, 1 row affected (0.001 sec)

MariaDB [(none)]> use test2;
Database changed
MariaDB [test2]> select @@character_set_connection;
+----------------------------+
| @@character_set_connection |
+----------------------------+
| utf8                       |
+----------------------------+
1 row in set (0.000 sec)

MariaDB [test2]> select @@collation_connection;
+------------------------+
| @@collation_connection |
+------------------------+
| utf8_general_ci        |
+------------------------+
1 row in set (0.000 sec)

MariaDB [test2]> create table c1 (a varchar(10))engine=columnstore;
Query OK, 0 rows affected (1.556 sec)

MariaDB [test2]> insert into c1 values ('abcde');
Query OK, 1 row affected (0.218 sec)

MariaDB [test2]> INSERT INTO c1 VALUES ('oe'),('ö');
Query OK, 2 rows affected (0.253 sec)
Records: 2  Duplicates: 0  Warnings: 0

MariaDB [test2]> select * from c1;
+-------+
| a     |
+-------+
| abcde |
| oe    |
| ö     |
+-------+
3 rows in set (0.050 sec)

MariaDB [test2]> select charset(a) from c1;
+------------+
| charset(a) |
+------------+
| utf8       |
| utf8       |
| utf8       |
+------------+
3 rows in set (0.006 sec)

MariaDB [test2]> select quote(a) from c1;
+----------+
| quote(a) |
+----------+
| 'abcde'  |
| 'oe'     |
| 'ö'      |
+----------+
3 rows in set (0.007 sec)

MariaDB [test2]> select hex(a) from c1;
+------------+
| hex(a)     |
+------------+
| 6162636465 |
| 6F65       |
| C3B6       |
+------------+
3 rows in set (0.007 sec)

```

**Review by LinuxJedi — APPROVED** [review]

File: utils/funcexp/func_quote.cpp [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    for (char ch : str)
+    {
+      new_length += get_esc_bit(bit_mask, (unsigned char) ch);
+    }
+
+    char *result = new char[new_length + 1];
+    char *to = result;
+
+    *to++ = '\'';
+    for (uint64_t i = 0; i < str.size(); i++, to++)
```
Comment (drrtuy) [inline utils/funcexp/func_quote.cpp]: You have two loops over the argument chars. Could you reduce a number of loops over the argument to one?

Comment (tntnatbry) [reply utils/funcexp/func_quote.cpp]: Changes made as discussed. Please review.

**Review by drrtuy — APPROVED** [review]

**Pull request merged**